### ISSUE TITLE
The json reporter does not print results of benchmarks

### ISSUE THREAD

**Issue by SimeonEhrig** [catchorg/Catch2#2879]
**Describe the bug**
The json reporter does not print results of benchmarks

**Expected result**

The xml reporter provides the benchmark results: [bench.xml.txt](https://github.com/user-attachments/files/15983475/bench.xml.txt)

The json reporter misses the benchmark results: [bench.json](https://github.com/user-attachments/files/15983464/bench.json)

**Reproduction steps**

1. compile the following code with catch2
2. run with `./bench -r json > bench.json`

```c++
#include <catch2/benchmark/catch_benchmark.hpp>
#include <catch2/catch_test_macros.hpp>

#include <cctype>

std::uint64_t Fibonacci(std::uint64_t number)
{
    return number < 2 ? 1 : Fibonacci(number - 1) + Fibonacci(number - 2);
}

TEST_CASE("Fibonacci")
{
    CHECK(Fibonacci(0) == 1);
    // some more asserts..
    CHECK(Fibonacci(5) == 8);
    // some more asserts..

    // now let's benchmark:
    BENCHMARK("Fibonacci 20")
    {
        return Fibonacci(20);
    };

    BENCHMARK("Fibonacci 25")
    {
        return Fibonacci(25);
    };

    BENCHMARK("Fibonacci 30")
    {
        return Fibonacci(30);
    };

    BENCHMARK("Fibonacci 35")
    {
        return Fibonacci(35);
    };
}
```

**Platform information:**
 - OS: **Ubuntu 22.04**
 - Compiler+version: **GCC 11.4**
 - Catch version: **v3.6.0**

**Comment by pavan591-0** [comment]
Hi @SimeonEhrig,

We want to work on this issue as part of our course project. Could you please assign it to me and, if possible, help us generate the issue?

Thank you.

**Comment by horenmar** [comment]
Yeah, JSON reporter is not finished, and any output aside the listings (`--list-*`) is a WIP skeleton. The reason it is merged is to support work on #2871 (which seems to have gotten stuck somewhere).

**Comment by SimeonEhrig** [comment]
> Hi @SimeonEhrig,
> 
> We want to work on this issue as part of our course project. Could you please assign it to me and, if possible, help us generate the issue?
> 
> Thank you.

Thanks. Unfortunately I cannot assign this issue to you, because I'm not a Catch 2 maintainer.
I'm not sure, what you mean with "generate the issue".

**Comment by pavan591-0** [comment]
Hi,

Our group has selected another issue and working on it.

Regards,
Pavan
